### PR TITLE
814: Added review screens for approve and wrap

### LINKED ISSUE #814: Add separate review screen for approving and wrapping

### PR DESCRIPTION
**Pull request by piersss**
Fixes #814 

For OTC, will do RFQ in separate PR.

Make flow with wrapping and approving demo:

https://github.com/airswap/airswap-web/assets/86911296/e9d1e3a5-7d99-468a-a2cc-135a5d4d976b

Take flow with approving demo:

https://github.com/airswap/airswap-web/assets/86911296/a0a14346-7ba2-4053-a9b5-4a02bd56a12d

All review screen variants:

Signer wrap:

wrap signer

Signer approve:

approve signer

Signer approve with already approved amount (but not sufficient):

approve signer with already approved amount

Sender wrap:

wrap sender

Sender approve:

approve sender

### CODE REVIEW

File: src/components/MakeOrderReview/MakeOrderReview.tsx [hunk 1]
```diff
@@ -2,59 +2,66 @@ import React, { FC, ReactElement, useMemo } from "react";
...
-import toRoundedNumberString from "../../../../helpers/toRoundedNumberString";
-import useInsufficientBalance from "../../../../hooks/useInsufficientBalance";
-import useShouldDepositNativeTokenAmountInfo from "../../../../hooks/useShouldDepositNativeTokenAmountInfo";
-import { OrderType } from "../../../../types/orderTypes";
-import { getTokenPairTranslation } from "../../helpers";
-import WalletLink from "../WalletLink/WalletLink";
+import { nativeCurrencyAddress } from "../../constants/nativeCurrency";
+import { getExpiryTranslation } from "../../helpers/getExpiryTranslation";
+import toRoundedNumberString from "../../helpers/toRoundedNumberString";
+import useShouldDepositNativeTokenAmountInfo from "../../hooks/useShouldDepositNativeTokenAmountInfo";
```
Comment (personnamedmike) [inline src/components/MakeOrderReview/MakeOrderReview.tsx:12]:
```suggestion
```

Comment (personnamedmike) [reply src/components/MakeOrderReview/MakeOrderReview.tsx:12]: var declared but never used

File: src/components/MakeWidget/helpers/index.ts
Comment (personnamedmike) [inline src/components/MakeWidget/helpers/index.ts:1]: unused variables in line 4 and 7

File: src/components/MakeWidget/subcomponents/ActionButtons/ActionButtons.tsx
Comment (personnamedmike) [inline src/components/MakeWidget/subcomponents/ActionButtons/ActionButtons.tsx:1]: Unused variable in line 4, `import { MakeWidgetState } from "../../MakeWidget";`

File: src/components/MakeWidget/subcomponents/MakeWidgetHeader/MakeWidgetHeader.styles.tsx
Comment (personnamedmike) [inline src/components/MakeWidget/subcomponents/MakeWidgetHeader/MakeWidgetHeader.styles.tsx:1]: Variable unused `import { MakeWidgetState } from "../../MakeWidget";`

File: src/components/MakeWidget/subcomponents/MakeWidgetHeader/MakeWidgetHeader.tsx
Comment (personnamedmike) [inline src/components/MakeWidget/subcomponents/MakeWidgetHeader/MakeWidgetHeader.tsx:1]: variables unused in lines 4 and 6
`import { WidgetHeader } from "../../../../styled-components/WidgetHeader/WidgetHeader";`
`import { MakeWidgetState } from "../../MakeWidget";`

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx
Comment (personnamedmike) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:1]: unused variable `selectTakeOtcStatus` in line 35

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 2]
```diff
@@ -110,8 +122,9 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
     signerAmount!
   );
   const hasApprovalPending = useApprovalPending(order.senderToken);
+  const wrappedNativeToken = useNativeWrappedToken(chainId);
   const orderTransaction = useSessionOrderTransaction(order.nonce);
-  const hasInsufficientAllowance = !useSufficientAllowance(
+  const { hasSufficientAllowance, readableAllowance } = useAllowance(
```
Comment (personnamedmike) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:127]: `readableAllowance` is unused

File: src/components/OrderReviewToken/OrderReviewToken.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { FC, ReactElement } from "react";
 
 import { TokenInfo } from "@airswap/types";
 
-import TokenLogo from "../../../TokenLogo/TokenLogo";
+import TokenLogo from "../TokenLogo/TokenLogo";
```
Comment (personnamedmike) [inline src/components/OrderReviewToken/OrderReviewToken.tsx:5]: `TokenLogo` is unused in the component.
As is `TokenInfo` in line 3

File: src/components/WrapReview/WrapReview.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import React, { FC, ReactElement, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+
+import { TokenInfo } from "@airswap/types";
+import { useToggle } from "@react-hookz/web";
+
+import { BigNumber } from "bignumber.js";
+
+import { nativeCurrencyAddress } from "../../constants/nativeCurrency";
```
Comment (personnamedmike) [inline src/components/WrapReview/WrapReview.tsx:9]: `nativeCurrencyAddress` variable unused in component

File: src/hooks/useAllowance.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { useMemo } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
```
Comment (personnamedmike) [inline src/hooks/useAllowance.ts:1]: `useCallback` and `useMemo` are unused.

Also line 3, `import { WETH }` is unused

File: src/hooks/useShouldDepositNativeTokenAmountInfo.ts
Comment (personnamedmike) [inline src/hooks/useShouldDepositNativeTokenAmountInfo.ts:1]: Component has unused variable `WETH` in line 3

**Review by personnamedmike — APPROVED** [review]
All the code looks good, except there are unused variables in several components

**Review by dmosites — APPROVED** [review]

**Pull request merged**